### PR TITLE
EO Clarifications

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #960 #822 

**Proposed Changes:**

1. Updated eo bands descriptions for more clarity
2. Put in h4's for better linking

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] a CHANGELOG entry is not required.

### CODE REVIEW

**Review by m-mohr — CHANGES_REQUESTED** [review]
Just requesting the required JSON Schema changes from me here... ;-)

File: extensions/eo/README.md [hunk 1]
```diff
@@ -28,16 +28,34 @@ For defining view geometry of data, it is strongly recommended to use the [`view
...
+on the `eo:bands` of each asset. An Item is only allowed to use `eo:bands` in its Properties if it has at least one asset with 
+a defined bands array.
+
+**NOTE**: In STAC versions 0.9.x and prior, `eo:bands` could only be used by an Asset putting the the Band Object definitions in an Item Properties and referencing these via array index. 1.0.0-beta.1 introduced the current behavior.
+
+#### eo:cloud_cover
+
+ Estimate of cloud cover as a percentage (0-100) of the entire scene. If not available, the field should not be provided. Generally, this value should be used in Item Properties rather than Item Assets, as an Item from an electro-optical source is a single snapshot of the Earth, so the cloud cover value would apply to all assets. 
 
 ### Band Object
```
Comment (m-mohr) [inline extensions/eo/README.md:60]: The Band Object doesn't require anything, thus the following is valid:

```
{
  href: 'example.tiff',
  eo:bands: [
    {},
    {}
  ]
}
```

I think we should either require a specific field or require at least one property to be available.

Comment (m-mohr) [reply extensions/eo/README.md:60]: I've updated the README and JSON Schema accordingly.

**Review by m-mohr — APPROVED** [review]
I've updated the JSON Schema (that looks/is very complicated now and I hope it works in all cases) and also added that at least one property is required in the Band Object and that the eo:bands array must contain at least one element if given.

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**